### PR TITLE
Produce pkg-config file under cmake

### PR DESCRIPTION
**Pull request by Sineaggi**
Produce a libglog.pc file that is identical to the one autotools generates and installs.

### CODE REVIEW

**Comment by Sineaggi** [comment]
Signed

**Comment by ilovezfs** [comment]
@shinh @ukai any chance of getting this merged? I just stumbled onto this PR because our switch to the CMake-based build in Homebrew apparently broke the build of supersonic, which looks for glog's pkg-config file.

File: CMakeLists.txt [hunk 1]
```diff
@@ -600,6 +600,17 @@ if (gflags_FOUND)
   set (gflags_DEPENDENCY "find_dependency (gflags ${gflags_VERSION})")
 endif (gflags_FOUND)
 
+if (OS_WINDOWS)
+  set (PKGCONFIG_INSTALL_DIR)
+else ()
+  set (PKGCONFIG_INSTALL_DIR ${_glog_CMake_LIBDIR}/pkgconfig)
+endif ()
```
Comment (sergiud) [inline CMakeLists.txt]: Please do not leave the conditions empty.

File: cmake/package.pc.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: libglog
+Description: Google Log (glog) C++ logging framework
+Version: @GLOG_VERSION@
+Libs: -L${libdir} -lglog
+Cflags: -I${includedir}
```
Comment (sergiud) [inline cmake/package.pc.in]: The compile definitions seem to be missing.

File: cmake/package.pc.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
```
Comment (sergiud) [inline cmake/package.pc.in]: I would rename `package.pc.in` to `libglog.pc.in`.

**Comment by Leif-W** [comment]
Problem 1) Has this been merged?  Any chance of that happening?  Pull request and CLAs signed over 18 months ago.  Checks completed over 14 months ago.  Issue remained open in limbo since?  This sort of neglect is a prime example favoring the complete removal of ALL Google's code from this and all future projects.

Problem 2) A missing pkg-config file in MSYS2's MINGW libglog v0.3.5 package has caused failure to build 3rd party software.  Not sure if this patch fixes it, or if v0.4.0 fixes the missing .pc file, or if it is related to some MSYS2/MINGW interaction.

**Comment by sergiud** [comment]
Superseded by #483.

**Pull request closed without merging**